### PR TITLE
feat: support models with comments

### PR DESCRIPTION
**Pull request by rhamzeh**
## Description

Note: Some go validations tests are expected to fail as we have not added that code yet

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/go/transformer/dsltojson.go [hunk 1]
```diff
@@ -372,7 +372,17 @@ func (c *OpenFgaDslErrorListener) SyntaxError(recognizer antlr.Recognizer, offen
 func ParseDSL(data string) (*OpenFgaDslListener, *OpenFgaDslErrorListener) {
 	cleanedLines := []string{}
 	for _, line := range strings.Split(data, "\n") {
-		cleanedLines = append(cleanedLines, strings.TrimRight(line, " "))
+		cleanedLine := ""
```
Comment (ewanharris) [inline pkg/go/transformer/dsltojson.go:375]: Is this something that could feasibly be moved to the lexer/parser?

**Review by ewanharris — APPROVED** [review]

File: pkg/go/transformer/dsltojson.go (hunk 1 above)
Comment (rhamzeh) [reply pkg/go/transformer/dsltojson.go:375]: Ideally, yeah. But we've had a hard time with that b/c `user#follower` is valid syntax that should be parsed.

Comment (rhamzeh) [reply pkg/go/transformer/dsltojson.go:375]: Ideally, yeah. But we've had a hard time with that b/c `user#follower` is valid syntax that should be parsed.

Comment (ewanharris) [reply pkg/go/transformer/dsltojson.go:375]: Thanks for the context!

**Pull request merged**